internal/resources/slo/resource_slo.go

```diff
@@ -298,19 +298,42 @@ func packSloResource(d *schema.ResourceData) (gapi.Slo, error) {
 		tflabels = packLabels(labels)
 	}
 
-	alerting := d.Get("alerting").([]interface{})
-	if len(alerting) > 0 {
-		alert := alerting[0].(map[string]interface{})
-		tfalerting = packAlerting(alert)
+	alerting, alertingOK := d.GetOk("alerting")
+
+	// means that the Alerting field on the Terraform State File was defined by the user
+	// it is either populated or is an empty block {}
+	if alertingOK {
+		alertData := alerting.([]interface{})
+
+		// if the Alerting field is an empty block, alertData[0] has a value of nil
+		if alertData[0] != nil {
+			// only pack the Alerting TF fields if the user populates the Alerting field with blocks
+			alert := alertData[0].(map[string]interface{})
+			tfalerting = packAlerting(alert)
+		}
+
+		slo := gapi.Slo{
```

Comment:
Instead of an early return here, consider just modifying the Slo object and having a single return for a no- error execution. I think it reduces the mental burden of modifying it later:
```go
slo := gapi.Slo{
	UUID:        d.Id(),
	Name:        tfname,
	Objectives:  tfobjective,
	Query:       tfquery,
	Alerting:    nil,
	Labels:      tflabels,
}

if alerting, ok := d.GetOk("alerting"); ok {
  ...
  tfAlerting := packAlerting(alert)
  slo.Alerting = &Alerting
}

return slo, nil
```

Reply:
Same here.